### PR TITLE
Add common vocabulary terms

### PR DESCRIPTION
**Pull request by RenaudWasTaken**
Hello!

After discussions with different people, I think we landed on the following definitions.
Credits to @mikebrow for a lot of the definitions :) !

/cc @mikebrow @mrunalp 
Signed-off-by: Renaud Gaubert <rgaubert@nvidia.com>

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -51,10 +51,13 @@ B. Container Runtime
...
 - Fields in CDI structures are required unless specifically marked optional.
 
-For the purposes of this proposal, we define two terms very specifically:
+For the purposes of this proposal, we define the following terms:
 - _container_ can be considered synonymous with a [Linux _network namespace_][namespaces].
-  What unit this corresponds to depends on a particular container runtime implementation. 
-- _device_ which will refer to an actual hardware device.
+  What unit this corresponds to depends on a particular container runtime implementation.
+- _device_ which refers to an actual hardware device.
+- _runtime engine_ which refers to the component that ingests the OCI specification and creates a container, e.g: runc, crun, ...
```
Comment (adrianchiris) [inline SPEC.md:58]: +1 on the additions !
i think its great to standardize terminology of the different components.

As a personal experience i've seen several terms being thrown around these components often unclear to what exactly they refer to.

Comment (mikebrow) [inline SPEC.md:58]: runtime engine is a component that creates a container from a OCI Runtime Specification and a rootfs directory.

File: SPEC.md [hunk 2]
```diff
@@ -51,10 +51,13 @@ B. Container Runtime
 - The device configuration includes mandatory fields such as "name" and "version".
 - Fields in CDI structures are required unless specifically marked optional.
 
-For the purposes of this proposal, we define two terms very specifically:
+For the purposes of this proposal, we define the following terms:
 - _container_ can be considered synonymous with a [Linux _network namespace_][namespaces].
```
[the remark this answers is not in the document]
Comment (mikebrow) [reply SPEC.md:55]: - _container_ is an instance of code running in an isolated execution environment specified by the OCI image and runtime specifications

File: SPEC.md [hunk 3]
```diff
@@ -51,10 +51,13 @@ B. Container Runtime
...
-  What unit this corresponds to depends on a particular container runtime implementation. 
-- _device_ which will refer to an actual hardware device.
+  What unit this corresponds to depends on a particular container runtime implementation.
+- _device_ which refers to an actual hardware device.
+- _runtime engine_ which refers to the component that ingests the OCI specification and creates a container, e.g: runc, crun, ...
+- _container runtime_ which refers to the higher level component users tend to interact with, e.g: docker, podman, ...
```
Comment (mikebrow) [inline SPEC.md:59]: interact with for managing containers... may also include lower level components that implement management of containers and pods (sets of containers).

File: SPEC.md [hunk 4]
```diff
@@ -51,10 +51,13 @@ B. Container Runtime
...
+- _container runtime interface integration_ which refers to the component that kubelet talks to and implements the Container Runtime Interface (CRI), e.g: containerd+cri, cri-o, ...
```
Comment (mikebrow) [inline SPEC.md:60]: split this one in two... one for the grpc interface client & server .. and one for the CRI integration - meeting the server side of the CRI interface for managing pods, containers, and container images on a node.

Comment (RenaudWasTaken) [reply SPEC.md]: Addressed

File: SPEC.md (hunk 1 above)
Comment (RenaudWasTaken) [reply SPEC.md]: Addressed

File: SPEC.md (hunk 3 above)
Comment (RenaudWasTaken) [reply SPEC.md]: Addressed

File: SPEC.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (RenaudWasTaken) [reply SPEC.md]: Addressed

**Comment by mrunalp** [comment]
this looks fine to me :+1:

**Pull request merged**